packages/shorebird_cli/test/src/commands/patch/android_patcher_test.dart

```diff
@@ -436,6 +441,60 @@ Looked in:
           );
 
           expect(result, hasLength(Arch.values.length));
+          for (final bundle in result.values) {
+            expect(bundle.hashSignature, isNull);
+          }
+        });
+
+        group('when a private key is provided', () {
+          const testPrivateKeyContent = '''
+-----BEGIN PRIVATE KEY-----
+key
+-----END PRIVATE KEY-----
+''';
+          const mockSignature = 'mock-signature';
+
+          late File privateKey;
+
+          setUp(() {
+            privateKey = File(
+              p.join(
+                Directory.systemTemp.createTempSync().path,
+                'test-private.pem',
+              ),
+            )..writeAsStringSync(testPrivateKeyContent);
+
+            when(() => argResults['private-key-path'])
+                .thenReturn(privateKey.path);
+
+            when(
+              () => codeSigner.sign(
+                message: any(named: 'message'),
+                privateKeyPemFile: any(named: 'privateKeyPemFile'),
+              ),
+            ).thenReturn(mockSignature);
+          });
+
+          test('returns patch artifact bundles with proper hash signatures',
+              () async {
+            final result = await runWithOverrides(
+              () => patcher.createPatchArtifacts(
+                appId: 'appId',
+                releaseId: 0,
+                releaseArtifact: File('release.aab'),
+              ),
+            );
+
+            const expectedSignatures = [
+              mockSignature,
+              mockSignature,
+              mockSignature,
+            ];
+
+            final signatures =
+                result.values.map((bundle) => bundle.hashSignature).toList();
```

Comment:
Might be useful to verify that we're generating a signature per hash? This reads like we might be using the same signature for every hash, which would (in most cases) not be right.